### DISCUSSION TITLE
[systemd question] How does the rtorrent service start as the `rtorrent` user?

### DISCUSSION THREAD

**Discussion by kartikynwa** [MichaIng/DietPi#7113] [Q&A] [answered]
This is what `/etc/systemd/system/rtorrent.service` looks like:

```ini
[Unit]
Description=rTorrent (DietPi)
Wants=network-online.target
After=network-online.target remote-fs.target

[Service]
User=rtorrent
ExecStart=/usr/bin/rtorrent

[Install]
WantedBy=multi-user.target
```

I can't see how it knows that it has to start as the `rtorrent` user. Yet it does. What am I missing here? I don't know systemd well.

**Comment by kartikynwa** [comment, marked as answer]
OMG I am an idiot. It's right there on line 7.

**Reply by MichaIng** [reply]
The official docs are not bad to get an overview and details about available systemd directives: https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#User=

It is split a bit annoyingly (IMO) across multiple pages, like [systemd.unit](https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html) for very general directives and mostly (but not only) those of the `[Unit]` section, the `systemd.exec` page I liked above for directives which have a direct effect on `Exec*` directives (like with which user/group it is executed), but without `Exec*` directives themselves, and the [systemd.service](https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html) page with other directives from the `[Service]` section, with a more indirect effect on how things are started, and the `Exec*´ directives themselves. Cycling through those 3 pages should answer most questions about how to achieve things and which directives/value means what.

And in case you plan to change the user, at best do not edit the service files themselves, but create a drop-in config, to override individual values. E.g.:
```sh
mkdir /etc/systemd/system/rtorrent.service.d
cat << '_EOF_' > /etc/systemd/system/rtorrent.service.d/myuser.conf
[Service]
User=myuser
_EOF_
systemctl daemon-reload
systemctl restart rtorrent
```
